src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -609,7 +652,7 @@ private static String getStringInsideTag(String result, DetailNode detailNode) {
      */
     private static String getFirstSentence(DetailNode ast) {
         final StringBuilder result = new StringBuilder(256);
-        final String periodSuffix = PERIOD + ' ';
+        final String periodSuffix = DEFAULT_PERIOD + ' ';
```

Comment:
It might be unrelated bug but this should be `period`
As in Japan language period is different symbol.

Reply:
Please create Input to catch this problem and prove that it will not happen.

Reply:
Yes, this is a bug, but we have a problem here.
If the locale is set to English then Javadoc will only consider `. ` as a period, eg-
```
/**                                               
 * This is 1.0 version. There is no other version.
 */                                               
```
The summary will be `This is 1.0 version.` and not `This is 1.`, this logic is not transferable to the Japanese language (according to google translate). There the period symbol is `。` (yes it has that extra space already associated with it), so the `period + ' '` logic won't work for it.
eg-
```
/**                            
 * これは1.0バージョンです。他のバージョンはありません。
 */                            
```
There is no extra space after the `period` symbol, the period symbol itself consists of that extra space. The only solution I see is to make a switch for `period` values and adjust logic based on that. We might like to research it a bit as [java supports many locales](https://www.oracle.com/java/technologies/javase/jdk8-jre8-suported-locales.html).


`. ` = `.` +  ` `
`。` = `。`

Reply:
I added logic for the Japanese language.

Reply:
The pitest will fail until #11407, I feel it would be better to fix this in #11407 and not include this in the current PR.

Reply:
>If the locale is set to English then Javadoc will only consider . as a period

We do not look at locale, period symbol is property of Check. Japanese people set it explicitly.

Reply:
>yes it has that extra space already associated with it)

We can do space check only if period character is not unicode.

Reply:
Yes, let's keep this separate, will add this issue to #11407